pkg/controllers/repositoryserver/handler.go

```diff
@@ -0,0 +1,297 @@
+// Copyright 2022 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package repositoryserver
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/jpillora/backoff"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	"github.com/kanisterio/kanister/pkg/kopia/command/storage"
+
+	crv1alpha1 "github.com/kanisterio/kanister/pkg/apis/cr/v1alpha1"
+	"github.com/kanisterio/kanister/pkg/kube"
+	"github.com/kanisterio/kanister/pkg/poll"
+)
+
+type RepoServerHandler struct {
+	Ctx                     context.Context
+	Req                     ctrl.Request
+	Logger                  logr.Logger
+	Reconciler              *RepositoryServerReconciler
+	KubeCli                 kubernetes.Interface
+	RepositoryServer        *crv1alpha1.RepositoryServer
+	RepositoryServerSecrets repositoryServerSecrets
+}
+
+func (h *RepoServerHandler) CreateOrUpdateOwnedResources() error {
+	svc, err := h.reconcileService()
+	if err != nil {
+		return err
+	}
+	if err = h.getSecretsFromCR(); err != nil {
+		return err
+	}
+	pod, err := h.reconcilePod(svc)
+	if err != nil {
+		return err
+	}
+	if err := h.waitForPodReady(pod); err != nil {
+		return err
+	}
+	if err := h.createKopiaRepository(); err != nil {
+		return err
+	}
+	if err := h.startRepoProxyServer(context.Background()); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (h *RepoServerHandler) reconcileService() (*corev1.Service, error) {
+	repoServerNamespace := h.RepositoryServer.Namespace
+	serviceName := h.RepositoryServer.Status.ServerInfo.ServiceName
+	svc := &corev1.Service{}
+	h.Logger.Info("Check if Service resource exists. If exists, reconcile with CR spec")
+	err := h.Reconciler.Get(h.Ctx, types.NamespacedName{Name: serviceName, Namespace: repoServerNamespace}, svc)
+	if err == nil {
+		return h.updateService(svc)
+	}
+	if !apierrors.IsNotFound(err) {
+		return nil, err
+	}
+	h.Logger.Info("Service resource not found. Creating new service")
+	svc, err = h.createService(repoServerNamespace)
+	if err != nil {
+		return nil, err
+	}
+	h.Logger.Info("Update serviceName in RepositoryServer /status")
```

Comment:
I suppose these logs are temporary only?

Reply:
No there are not temporary logs, this is to let user know what are the things that are happening in the controller

Reply:
```suggestion
	h.Logger.Info("Update ServiceName in RepositoryServer /status")
```